### PR TITLE
CEDS-1404 Use only document reference without parts  

### PR DESCRIPTION
**Pull request by bambuchaAdm**

### CODE REVIEW

File: test/forms/declaration/DocumentsProducedSpec.scala [hunk 1]
```diff
@@ -68,31 +66,18 @@ class DocumentsProducedSpec extends WordSpec with MustMatchers with DocumentsPro
 
       "provided with Document Identifier and Part" which {
```
Comment (MaciejRewera) [inline test/forms/declaration/DocumentsProducedSpec.scala]: Probably should be without `Part`

File: app/views/components/buttons/remove_button.scala.html [hunk 2]
```diff
@@ -18,4 +18,4 @@
 
 @(value: Option[Any] = None, label: String = "site.remove", buttonClass: String = "button--secondary")(implicit messages: Messages)
 
-<button id="remove" class="@buttonClass" name="@Remove.toString" @value.map{v => value="@{v.toString}"}>@messages(label)</button>
\ No newline at end of file
+<button class="remove @buttonClass" name="@Remove.toString" @value.map{v => value="@{v.toString}"}>@messages(label)</button>
```
Comment (SteveSugden) [inline app/views/components/buttons/remove_button.scala.html]: What is the "remove" css class?  And should this button not have an id?

File: test/views/declaration/spec/UnitViewSpec.scala [hunk 3]
```diff
@@ -39,6 +42,27 @@ class UnitViewSpec extends UnitSpec with ViewMatchers {
       "error.summary.text"
     )
   }
+
+  def messagesKey(key: String): BeMatcher[String] = new MessagesKeyMatcher(key)
+}
+
+class MessagesKeyMatcher(key: String) extends BeMatcher[String] {
```
Comment (SteveSugden) [inline test/views/declaration/spec/UnitViewSpec.scala]: Is this used anywhere in this PR?  Can't see it.

File: app/views/declaration/documents_produced.scala.html [hunk 4]
```diff
@@ -45,34 +42,32 @@
...
-                        <td>@components.buttons.remove_button(value = Some(item.toJson))</td>
+                        <td scope="row" class="document-type">@item.documentTypeCode</td>
+                        <td class="document-identifier">@item.documentIdentifier</td>
+                        <td class="document-status">@item.documentStatus</td>
+                        <td class="document-status-reason">@item.documentStatusReason</td>
+                        <td class="document-issuing-authority">@item.issuingAuthorityName</td>
+                        <td class="date-of-validity">@item.dateOfValidity</td>
+                        <td class="measurement-unit">@item.documentWriteOff.map(_.measurementUnit)</td>
+                        <td class="document-quantity">@item.documentWriteOff.map(_.documentQuantity)</td>
+                        <td class="actions">@components.buttons.remove_button(value = Some(item.toJson))</td>
```
Comment (SteveSugden) [inline app/views/declaration/documents_produced.scala.html]: What are all these css classes for?

File: app/forms/declaration/additionaldocuments/DocumentsProduced.scala [hunk 5]
```diff
@@ -71,7 +71,7 @@ object DocumentsProduced {
       documentIdentifierKey -> optional(
         text().verifying(
           "supplementary.addDocument.documentIdentifier.error",
-          nonEmpty and isAlphanumeric and noLongerThan(35)
+          nonEmpty and isAlphanumericWithAllowedSpecialCharacters and noLongerThan(35)
```
Comment (SteveSugden) [inline app/forms/declaration/additionaldocuments/DocumentsProduced.scala]: :+1

File: app/views/components/buttons/remove_button.scala.html (hunk 2 above)
Comment (bambuchaAdm) [reply app/views/components/buttons/remove_button.scala.html]: Those buttons are used multiple time on site and break uniqueness of id there - remove is just access class same as id but its legal to have multiple tags to share it

File: app/views/declaration/documents_produced.scala.html (hunk 4 above)
Comment (bambuchaAdm) [reply app/views/declaration/documents_produced.scala.html]: Mostly for navigation over table with css selector and not depends on index in table

File: test/views/declaration/spec/UnitViewSpec.scala (hunk 3 above)
Comment (bambuchaAdm) [reply test/views/declaration/spec/UnitViewSpec.scala]: Probably github hide diff on view spec test/views/declaration/DocumentsProducedViewSpec.scala - whole view spec have been rewritten by me and it was used there

File: app/views/declaration/documents_produced.scala.html (hunk 4 above)
Comment (bambuchaAdm) [reply app/views/declaration/documents_produced.scala.html]: Accesses in tests - classes are much more resilience for changes then id or indexes

**Pull request merged**